src/main/java/org/fever/filecreator/ClassArgumentFetcher.java

```diff
@@ -0,0 +1,100 @@
+
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class ClassArgumentFetcher {
+    private static final int SELF_INDEX_IN_PARAMETER_LIST = 0;
+    private static final ResolutionCache.State resolutionCacheState = ApplicationManager.getApplication().getService(ResolutionCache.class).getState();
+
+    public static Collection<IdentifierItem> getFqnOfInitArguments(PyFile sourceCodeFile) {
+        List<IdentifierItem> identifiers = new ArrayList<>();
+
+        PyClass pyClass = sourceCodeFile.getTopLevelClasses().get(0);
+        TypeEvalContext context = TypeEvalContext.codeCompletion(
+                sourceCodeFile.getProject(),
+                sourceCodeFile.getContainingFile()
+        );
+        Map<PyParameter, @Nullable PyClass> initParamsToClasses = getClassesFromInitParams(pyClass, context);
+
+        for (Map.Entry<PyParameter, @Nullable PyClass> entry : initParamsToClasses.entrySet()) {
+            PyParameter parameter = entry.getKey();
+            PyClass parameterClass = entry.getValue();
+
+            if (parameterClass == null) {
+                @Nullable String identifier = getIdentifierFromCache(sourceCodeFile, parameter);
+                identifiers.add(new IdentifierItem(identifier, null, parameter));
+                continue;
+            }
+
+            Collection<PyClass> allImplementations = PyClassInheritorsSearch.search(parameterClass, false).findAll();
+            if (allImplementations.isEmpty()) {
+                PsiFile dependencyInjectionFile = GotoPypendencyOrCodeHandler.getPypendencyDefinition(parameterClass.getContainingFile());
+                String parameterClassIdentifier = IdentifierExtractor.extractIdentifierFromDIFile(dependencyInjectionFile);
+                if (parameterClassIdentifier != null) {
+                    identifiers.add(new IdentifierItem(parameterClassIdentifier, parameterClass, parameter));
+                    continue;
+                }
+                identifiers.add(new IdentifierItem(null, parameterClass, parameter));
+                continue;
+            }
+
+            for (PyClass implementation : allImplementations) {
+                PsiFile dependencyInjectionFile = GotoPypendencyOrCodeHandler.getPypendencyDefinition(implementation.getOriginalElement().getContainingFile());
+                String implementationIdentifier = IdentifierExtractor.extractIdentifierFromDIFile(dependencyInjectionFile);
+                IdentifierItem item = new IdentifierItem(implementationIdentifier, parameterClass, parameter);
+                if (!identifiers.contains(item)) {
+                    identifiers.add(item);
+                }
+            }
+        }
+
+        return identifiers;
+    }
+
+    private static String getIdentifierFromCache(PyFile sourceCodeFile, PyParameter parameter) {
+        Matcher argumentTypeMatcher = Pattern.compile("\\w+:\\s+(\\w+)").matcher(parameter.getText());
```

Comment:
Also support strings like `logger:Logger` (without the whitespace after the `:`)
```suggestion
        Matcher argumentTypeMatcher = Pattern.compile("\\w+:\\s*(\\w+)").matcher(parameter.getText());
```